core/src/databases/database.rs

```diff
@@ -77,15 +78,21 @@ pub async fn execute_query(
             }
         }
         Ok(TableType::Local) => {
-            execute_query_on_transient_database(
-                &tables
-                    .into_iter()
-                    .map(|t| LocalTable::from_table(t))
-                    .collect::<Result<Vec<_>>>()?,
-                store,
-                query,
+            let mut local_tables = tables
+                .into_iter()
+                .map(|t| LocalTable::from_table(t))
+                .collect::<Result<Vec<_>>>()?;
+
+            // Load the schema for each table.
+            // If the schema cache is stale, this will update it in place.
+            try_join_all(
+                local_tables
+                    .iter_mut()
+                    .map(|t| t.schema(store.clone(), databases_store.clone())),
```

Comment:
So we always end up reloading the schema for all tables each time we do a query. Is there a perf concern here?

Reply:
How is that computed? load from DB? recompute?